### PR TITLE
Fix WebRTC visualizer bug

### PR DESCRIPTION
**Pull request by farzadab**
Bugfix:
- state changes to speaking before `outAnalyzer` gets set (React doesn't know about underlying object change)
- setting `audioContext` on the `mediaStream` of LiveKit remote track

### CODE REVIEW

File: packages/voice/src/app/agent/chat.tsx [hunk 1]
```diff
@@ -675,8 +675,10 @@ export class WebRtcChatManager implements ChatManager {
     const audioTrack = track as RemoteAudioTrack;
     audioTrack.on(TrackEvent.AudioPlaybackStarted, () => console.log(`[chat] audio playback started`));
     audioTrack.on(TrackEvent.AudioPlaybackFailed, (err) => console.error(`[chat] audio playback failed`, err));
+    audioTrack.setAudioContext(this.audioContext);
```
Comment (juberti) [inline packages/voice/src/app/agent/chat.tsx:678]: OOC, why is this needed?

File: packages/voice/src/app/agent/chat.tsx [hunk 2]
```diff
@@ -685,6 +687,11 @@ export class WebRtcChatManager implements ChatManager {
       console.debug(`[chat] worker RTT: ${elapsed_ms.toFixed(0)} ms`);
     } else if (data.type === 'state') {
       const newState = data.state;
+      if (newState === 'speaking' && this.outAnalyzer === undefined) {
```
Comment (juberti) [inline packages/voice/src/app/agent/chat.tsx:690]: should this be `ChatManagerState.SPEAKING`?

File: packages/voice/src/app/agent/chat.tsx [hunk 3]
```diff
@@ -675,8 +675,10 @@ export class WebRtcChatManager implements ChatManager {
     const audioTrack = track as RemoteAudioTrack;
     audioTrack.on(TrackEvent.AudioPlaybackStarted, () => console.log(`[chat] audio playback started`));
     audioTrack.on(TrackEvent.AudioPlaybackFailed, (err) => console.error(`[chat] audio playback failed`, err));
+    audioTrack.setAudioContext(this.audioContext);
     audioTrack.attach(this.audioElement);
     this.outAnalyzer = new StreamAnalyzer(this.audioContext, track.mediaStream!);
+    this.changeState(ChatManagerState.SPEAKING);
```
Comment (juberti) [inline packages/voice/src/app/agent/chat.tsx:681]: Probably only want to do this in the case you mention below, right?

**Review by juberti — COMMENTED** [review]
Nice finds, this mostly makes sense to me. Noted a couple nits.

File: packages/voice/src/app/agent/chat.tsx (hunk 1 above)
Comment (farzadab) [reply packages/voice/src/app/agent/chat.tsx:678]: The real answer is I was just playing with it and saw the function exists, thought maybe I should connect the two objects because they seemed to be coming from different places ... so basically just trial and error 😅

File: packages/voice/src/app/agent/chat.tsx (hunk 3 above)
Comment (farzadab) [reply packages/voice/src/app/agent/chat.tsx]: Updated.

File: packages/voice/src/app/agent/chat.tsx (hunk 1 above)
Comment (juberti) [reply packages/voice/src/app/agent/chat.tsx:680]: OK... would be good to try to understand what is happening internally and add a comment indicating that as it's not clear at all to me how setting this context affects our own visualizer objects

File: packages/voice/src/app/agent/chat.tsx [hunk 4]
```diff
@@ -571,7 +573,7 @@ export class WebRtcChatManager implements ChatManager {
   }
   warmup() {
     const isLocalHost = window.location.hostname === 'localhost';
-    const url = this.params.webrtcUrl || (!isLocalHost ? 'wss://wsapi.fixie.ai' : 'ws://localhost:8100');
+    const url = this.params.webrtcUrl || (isLocalHost ? 'wss://wsapi.fixie.ai' : 'ws://localhost:8100');
```
Comment (juberti) [inline packages/voice/src/app/agent/chat.tsx:576]: this doesn't seem right (localhost is set up to point at a local voice server)

**Review by juberti — APPROVED** [review]
LG but see comment about `setAudioContext`

File: packages/voice/src/app/agent/chat.tsx (hunk 1 above)
Comment (farzadab) [reply packages/voice/src/app/agent/chat.tsx:680]: Sorry I didn't get to figure it out. I left a TODO for when I come back to it.
